templates/admin-module.js

```diff
@@ -1,78 +1,81 @@
 const chalk = require('chalk')
+const path = require('path')
 const log = require('../lib/logger')
 
+const {input, checkbox} = require('../lib/inputs')
+
+const templatePath = 'templates/admin-module'
 const description = 'Generate an Angular module for @colmena/admin'
 
-const prompts = [{
-  type: 'input',
-  name: 'moduleName',
-  message: 'Module Name:',
-  validate: (value) => {
-    if (/.+/.test(value)) {
-      return true
-    }
-    return 'Module Name is required'
-  },
-}, {
-  type: 'input',
-  name: 'modelName',
-  message: 'Primary Model:',
-  validate: (value) => {
-    if (/.+/.test(value)) {
-      return true
-    }
-    return 'Primary Model is required'
-  },
-}, {
-  type: 'checkbox',
-  name: 'moduleItems',
-  message: 'Select the items you want to include:',
-  choices: [
+const prompts = [
+  input('moduleName', 'Module Name:', true),
+  input('modelName', 'Primary Name:', true),
+  checkbox('moduleItems', 'Select the items you want to include:', [
     {name: 'Service', value: 'service', checked: true},
     {name: 'Resolvers', value: 'resolvers', checked: true},
     {name: 'Routes', value: 'routes', checked: true},
     {name: 'Containers', value: 'containers', checked: true},
     {name: 'Components', value: 'components', checked: true},
-  ],
-}]
+  ]),
+]
 
 const actions = (data) => {
-  const actions = [
-    {
-      type: 'add',
-      path: '{{ lowerCase moduleName }}/{{ dashCase moduleName }}.module.ts',
-      templateFile: 'templates/admin-module/module.txt',
+  const targetPath = '{{ lowerCase moduleName }}'
+  const moduleName = '{{ dashCase moduleName }}'
+
+  const actions = []
+
+  // Push and add action to the actions array, prepend target and template path
+  const addFile = (templateFile, ...targetFiles) => actions.push({
+    type: 'add',
+    templateFile: path.join(templatePath, templateFile),
+    path: path.join(targetPath, ...targetFiles),
+  })
```

Comment:
Pulling out the addFile method and prefixing the paths saves a lot of duplication.